db/migrate/20190708192323_fix_unserializable_notification_options.rb

```diff
@@ -0,0 +1,39 @@
+class FixUnserializableNotificationOptions < ActiveRecord::Migration[5.1]
+  class Notification < ActiveRecord::Base
+    belongs_to :notification_type, :class_name => "FixUnserializableNotificationOptions::NotificationType"
+  end
+
+  class NotificationType < ActiveRecord::Base
+  end
+
+  # only used for testing
+  class CloudVolume < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    invalid_options_ids = []
+    removed_records_ids = []
+    Notification.where.not(:options => nil).all.each do |n|
```

Comment:
note, I changed this back to `.all.each` instead of `.find_each` because the latter fails with the lab database with:

```
PG::SyntaxError: ERROR:  zero-length delimited identifier at or near """"
LINE 1: ..."."options" IS NOT NULL) ORDER BY "notifications"."" ASC LIM...
                                                             ^
: SELECT  "notifications".* FROM "notifications" WHERE ("notifications"."options" IS NOT NULL) ORDER BY "notifications"."" ASC LIMIT $1
```

🤷‍♂

Reply:
I think you can do just `.each`, no need for the `all` when you already have a relation.

Reply:
yeah, you're right... I was 😢 when `not(...).find_each` didn't work

Reply:
ok, fixed